packages/block/src/header.ts

```diff
@@ -132,6 +135,20 @@ export class BlockHeader {
       },
     ]
     defineProperties(this, fields, data)
+
+    if (this._common.hardforkIsActiveOnChain('dao')) {
+      // verify the extraData field.
+      const blockNumber = new BN(this.number)
+      const DAOActivationBlock = new BN(this._common.hardforkBlock('dao'))
+      if (blockNumber.gte(DAOActivationBlock)) {
+        const drift = blockNumber.sub(DAOActivationBlock)
+        if (drift.lten(DAO_ForceExtraDataRange)) {
+          if (!this.extraData.equals(DAO_ExtraData)) {
+            throw new Error("extraData should be 'dao-hard-fork'")
+          }
+        }
+      }
+    }
```

Comment:
Sorry that this didn't occur to me before, a bit analogue to other requests since all this DAO code is so special: can we extract this code block to a separate `_checkDAOForkExtraData()` (or similar) method since this code block is unproportionally to its importance bloating the constructor code so much right now.

Reply:
(with function placement optimally also very much at the end of the file along the other `_*` functions)

Reply:
Update: I will just add this myself along the review and then approve to streamline the process. 😄

Reply:
Could you also update the Changelog? 😄